cx/op_und.go

```diff
@@ -659,31 +659,14 @@ func opRead(prgrm *CXProgram) {
 	fp := prgrm.GetFramePointer()
 
 	out1 := expr.Outputs[0]
-	out1Offset := GetFinalOffset(fp, out1)
 
 	reader := bufio.NewReader(os.Stdin)
 	text, err := reader.ReadString('\n')
-	// text = strings.Trim(text, " \n")
-	text = strings.Replace(text, "\n", "", -1)
-	text = strings.Replace(text, "\r", "", -1)
-
 	if err != nil {
-		panic("")
-	}
-	byts := encoder.Serialize(text)
-	size := encoder.Serialize(int32(len(byts)) + OBJECT_HEADER_SIZE)
-	heapOffset := AllocateSeq(len(byts) + OBJECT_HEADER_SIZE)
-
-	var header = make([]byte, OBJECT_HEADER_SIZE)
-	for c := 5; c < OBJECT_HEADER_SIZE; c++ {
-		header[c] = size[c-5]
+		panic(err)
 	}
+	text = strings.Replace(text, "\n", "", -1)
+	text = strings.Replace(text, "\r", "", -1)
```

Comment:
Hmm, I am doubtful about this.  Sometimes the user may want to actually enter \n or \r by using, say, shift-enter and they should be allowed to.  To remove \n\r from the end of the string is totally ok, though.

Reply:
hm, right. I thought that code was removing newlines from the beginning and the end (because that's done in the lexer, when reading... strings, I think). I assumed it was something that didn't need my attention and didn't check. I'll change it.

Reply:
Ah, in that case maybe we should fix it in the parser also.  But before doing that, let's check the docs for strings.Replace().